primefaces/src/main/java/org/primefaces/component/api/UIData.java

```diff
@@ -622,7 +625,18 @@ else if (component instanceof UIForm) {
                 restoreDescendantState(facet, context);
             }
         }
+    }
 
+    protected SavedState getSavedState(UIComponent component, FacesContext context) {
+        Map<String, SavedState> saved = (Map<String, SavedState>) getStateHelper().get(PropertyKeys.saved);
+        if (saved == null) {
+            return new SavedState();
```

Comment:
should we return a static empty SavedState?

Reply:
If the IT are passing I can change it to a static one.